sdk/rust/oak/src/lib.rs

```diff
@@ -43,6 +44,44 @@ pub struct Handle {
     id: u64,
 }
 
+/// Serialize a `Handle` as the invalid handle value.
```

Comment:
I don't think I really understand what this is for, but on the face of it it appears to be a dreadful footgun.

Will this allow users to serde-serialize structures holding `oak::Handle`s but will silently drop those handles?  So the type system tells the user that everything is hunky-dory and type-safe but is actually throwing things away under the covers?

(Or am I completely misunderstanding what's going on?)

Reply:
Yes, this allows structs containing `Handle`s to be serialized via serde, and it will silently convert them to invalid handles. Which is exactly what I do now in the example, in which I encode the command as bytes via bincode (using serde), and then manually pull out handles from the original struct, and put them back together on the other side.

FTR, I don't think this is any less type safe than creating `Handle` from a raw number?

I cannot think of a better solution that has the flexibility of serializing large structs with arbitrary handles using existing formats (bincode, json), but I'm happy to hear suggestions.

Obviously FIDL solves this problem (except that it uses its own type for handles, and we would still have to manually convert to and from it, or fork FIDL to use our type).

Reply:
(From discussion).

So I wondered if we could have a trait `Encodable` with a method `encode` to bytes and handles (like in `chat/common/src/lib.rs`) that re-uses `Serialize` where available.  So POD types become automatically `Encodable` (because they're `Serialize`able) and Oak Handle structs are explicitly `Encodable`, meaning that `struct`s that combine the two are `Encodable` &ndash; but are **not** `Serialize`able.  But I don't know enough about trait inheritance/borrowing to immediately know if this is feasible or not&hellip;

Reply:
Created #463 to track, thanks for writing this down!